src/code/dialogs/zoneDialog.js

```diff
@@ -97,18 +128,165 @@ const ZoneDialog = WDialog.extend({
           getSelectedOperation().removeZone(z.id);
         });
       }
+      if (z.points.length == 0) {
+        const addPoints = L.DomUtil.create("a", null, commandcell);
+        addPoints.textContent = " 🖍";
+        addPoints.href = "#";
+        L.DomEvent.on(addPoints, "click", (ev) => {
+          L.DomEvent.stop(ev);
+          this.ZonedrawHandler.zoneID = z.id;
+          this.ZonedrawHandler.enable();
+        });
+      } else {
+        const delPoints = L.DomUtil.create("a", null, commandcell);
+        delPoints.textContent = " 🧽";
+        delPoints.href = "#";
+        L.DomEvent.on(delPoints, "click", (ev) => {
+          L.DomEvent.stop(ev);
+          getSelectedOperation().removeZonePoints(z.id);
+        });
+      }
+      if (this.ZonedrawHandler && this.ZonedrawHandler._enabled) {
+        const stopDrawing = L.DomUtil.create("a", null, commandcell);
+        stopDrawing.href = "#";
+        stopDrawing.textcontent = " 🛑";
+        L.DomEvent.on(stopDrawing, "click", (ev) => {
+          L.DomEvent.stop(ev);
+          this.ZonedrawHandler.disable();
+        });
+      }
     }
 
-    const add = L.DomUtil.create("a", null, container);
+    /* const add = L.DomUtil.create("a", null, container);
     add.href = "#";
     add.textContent = "add zone";
     L.DomEvent.on(add, "click", (ev) => {
       L.DomEvent.stop(ev);
       getSelectedOperation().addZone();
     });
 
+    const assign = L.DomUtil.create("a", null, container);
+    assign.href = "#";
+    assign.textContent = "set markers to zones";
+    L.DomEvent.on(assign, "click", (ev) => {
+      L.DomEvent.stop(ev);
+      this.setMarkersToZones();
+    });
+    */
+
     return container;
   },
+
+  setMarkersToZones: function () {
```

Comment:
setMarkersToZones and setLinksToZones should probably be moved to uiCommands.js so other things can use them